app/services/budgetkey-item.ts

```diff
@@ -46,22 +46,41 @@ export class BudgetKeyItemService {
     Promise.reject(new Error('No layout for ' + path));
   }
 
-  getItemData(query: string, headersOrder: string[]): Promise<object> {
+  private budgetNumberFormatter(value: any)
+  {
+    return parseFloat(value).toLocaleString('he-IL',{ style: 'currency', currency: 'ILS' });
+  }
+  
+  private budgetLinkFormatter(value: string, hLink: string)
+  {
+    return '<a>' + value + '</a>';
+  }
+
+  getItemData(query: string, headersOrder: string[], formatters: string[]): Promise<object> {
     let url = 'http://next.obudget.org/api/query?query=' +
       encodeURIComponent(query);
     return new Promise<any>((resolve, reject) => {
       this.http.get(url)
         .map((res: any) => res.json())
         .subscribe(
           (res: any) => {
-            let items: object[] = [];
-            let rows = res.rows;
-            let headers = rows.length > 0 ? _.union(headersOrder, _.keys(_.first(rows))) : [];
+            let items: object[] = [], rows = res.rows, 
+            headers = rows.length > 0 ? _.union(headersOrder, _.keys(_.first(rows))) : [];
+            
             _.each(rows, (row) => {
               let newItem: any[] = [];
+              
               _.each(headers, (header) => {
-                newItem.push(row[header]);
+                let item = row[header];
+                
+                (typeof(item) != 'object') &&  _.each(formatters, (formatter) => {
```

Comment:
not sure if this will still be relevant after your refactor, but doing a double iteration (on all headers and on all formatters) is quite inefficient

Reply:
the checking for a type is kind of performance improvement...
Do you have another solution instead of checking each table item for its column name and comparing it with all the Formatter Keys?

Reply:
Not sure what's the problem if item is an object...

anyway, for another solution see my comment https://github.com/OpenBudget/budgetkey-app-generic-item/pull/41#discussion_r159365962 above